plugins/out_kinesis_streams/kinesis_api.c

```diff
@@ -97,6 +97,47 @@ static int init_put_payload(struct flb_kinesis *ctx, struct flush *buf,
     return -1;
 }
 
+/*
+ * Adjust the time_key output format according to time_key_format.
+ */
+static void get_time_format_string(char *str, size_t maxsize,
+                      const char *time_key_format, const struct tm *timeptr,
+                      size_t len, struct flb_time *tms){
+    int millisecond_length = ((int)((ceil(log10(abs((uint64_t) tms->tm.tv_nsec / 1000000)))+1)*sizeof(char)));
+    int nanosecond_length = ((int)((ceil(log10(abs((uint64_t) tms->tm.tv_nsec)))+1)*sizeof(char)));
+    char millisecond_string[millisecond_length];
+    char nanosecond_string[nanosecond_length];
```

Comment:
I'm actually not sure how this can compile?
arrays need to have a static length not a variable length

Reply:
I don't know myself either but from quick googling it sounds like this might be a C99 feature: https://www.ibm.com/docs/en/ent-metalc-zos/3.1?topic=SSSHGK_3.1.0/com.ibm.metalc.v3r1.langref/variable_length_arrays.htm

Unless I misread that

Reply:
I do not think we need a variable length array tho... milliseconds can only ever be 3 chars and nanoseconds 9 chars right? So we can use a constant expression here.